frontend/src/Router.tsx

```diff
@@ -169,14 +143,32 @@ export const routeAtom = atom(
     const config = get(appConfigAtom);
     return locationToRoute(config.root, location);
   },
-  (get, set, value: Route) => {
+  (get, set, value: Route, searchParams?: URLSearchParams) => {
     const appConfig = get(appConfigAtom);
     set(locationAtom, {
       pathname: appConfig.root + routeToPath(value),
+      searchParams: searchParams,
     });
   },
 );
 
+/**
+ * Check for actions in URL query params requiring a redirect
+ * Get route from path
+ * @returns Route
+ */
+const useRouteWithRedirect = (): Route => {
+  const location = useAtomValue(locationAtom);
+  const redirect = getRouteActionRedirection(location);
+
+  const [route, setRoute] = useAtom(routeAtom);
+  if (redirect) {
+    setRoute(redirect.route, redirect.searchParams);
```

Comment:
I think this should be in a useEffect, because you're not supposed to do state changes in the render?

Reply:
Kerry won't see this before Monday, sadly, and we have to land this PR today. @t3chguy could you help us out by giving advice?

Reply:
I can take care of finishing that up, alongside the server side of this

Reply:
Yes I believe it should be in a useEffect